### PR TITLE
WR#249863: Fix issue #2.

### PR DESCRIPTION
**Pull request by suankan**

### CODE REVIEW

File: lang/en/local_linkchecker_robot.php [hunk 1]
```diff
@@ -81,7 +81,7 @@
 $string['size'] = 'Size';
 $string['url'] = 'URL';
 $string['mimetype'] = 'Mime type';
-$string['broken'] = 'Broken URL';
+$string['brokenlinksandurls'] = 'Broken URL';
```
Comment (brendanheywood) [inline lang/en/local_linkchecker_robot.php]: This has broken stuff:

Invalid get_string() identifier: 'broken' or component 'local_linkchecker_robot'. Perhaps you are missing $string['broken'] = ''; in /var/www/vanilla-moodle/local/linkchecker_robot/lang/en/local_linkchecker_robot.php?
line 349 of /lib/classes/string_manager_standard.php: call to debugging()
line 6820 of /lib/moodlelib.php: call to core_string_manager_standard->get_string()
line 110 of /local/linkchecker_robot/report.php: call to get_string()

File: lang/en/local_linkchecker_robot.php [hunk 2]
```diff
@@ -81,7 +81,7 @@
 $string['size'] = 'Size';
 $string['url'] = 'URL';
 $string['mimetype'] = 'Mime type';
-$string['broken'] = 'Broken URL';
```
Comment (brendanheywood) [inline lang/en/local_linkchecker_robot.php:84]: also this should be something like 'Broken Links / URLS' - I think you did this and the commits are mixed up somewhere

File: index.php [hunk 3]
```diff
@@ -49,10 +49,18 @@
...
+$numurlsbroken = $DB->get_field_sql("SELECT COUNT(*)
                                  FROM {linkchecker_url}
                                 WHERE httpcode != ?", array('200') );
 
+$sqlnumpages = "SELECT COUNT (*)
+                         FROM {linkchecker_url}  b
+                    LEFT JOIN {linkchecker_edge} l ON l.b = b.id
+                    LEFT JOIN {linkchecker_url}  a ON l.a = a.id
+                    LEFT JOIN {course} c ON c.id = a.courseid
+                        WHERE b.httpcode != '200'";
```
Comment (brendanheywood) [inline index.php]: This whole sql doesn't look right, and definitely isn't producing correct numbers.

Off the top of my head: (please test yourself)

```
SELECT COUNT (*)
  FROM {linkchecker_url}  b
  JOIN {linkchecker_edge} l ON l.b = b.id
 WHERE b.httpcode != '200'";
```

File: lang/en/local_linkchecker_robot.php (hunk 1 above)
Comment (suankan) [inline lang/en/local_linkchecker_robot.php]: I've fixed this by reverting back all code references to as they were originally: get_string('broken'...). Then mapped this 'broken' string to 'Broken links / URLs' in language file.

File: lang/en/local_linkchecker_robot.php (hunk 2 above)
Comment (suankan) [inline lang/en/local_linkchecker_robot.php:84]: I've fixed this by reverting back all code references to as they were originally: get_string('broken'...). Then mapped this 'broken' string to 'Broken links / URLs' in language file.

File: index.php (hunk 3 above)
Comment (suankan) [inline index.php]: Yes this SQL gives correct result. I've amended it into the code.

**Pull request merged**